source/common/http/conn_manager_impl.h

```diff
@@ -528,7 +546,11 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     std::string downstream_address_;
     Optional<Router::RouteConstSharedPtr> cached_route_;
     DownstreamWatermarkCallbacks* watermark_callbacks_{nullptr};
+    uint32_t buffer_limit_{0};
     uint32_t high_watermark_count_{0};
+    bool encoder_filters_streaming_{true};
```

Comment:
Put in State struct above to use bitfield and also to keep state stuff together?

Reply:
These are all passed to various functions as booleans.  I think it's more clean to leave them as bools rather than get and set before and after function calls but let me know if you prefer that

Reply:
Sorry don't follow. `struct State` contains bools already. It's mostly just keeping related bools together (and allowing easier use of bit field for less memory). I don't think you need to do any get/set?

Reply:
It has bitfield style booleans.  That said there's absolutely no reason we can't move these over as normal bools.  consider it done (in about 5 minutes) :-)

Reply:
The bitfield style is just to use less memory. Why can't they just be bitfield style in there? That doesn't mean they are for setting/getting.